src/app/pages/wallets/wallet-details/pages-wallet-details-item.model.ts

```diff
@@ -7,11 +12,31 @@ export class WalletsDetailsTransaction {
   public readonly type: WalletTransactionType;
   public readonly amount: number;
 
-  public constructor(data: IWalletTransactionApiResponse) {
+  public constructor(data: IWalletTransactionItemData) {
     this.id = data.id;
     this.date = new Date(data.creationDate);
     this.description = data.description ?? null;
     this.type = data.type;
     this.amount = data.amount;
   }
+
+  public static create(data: Partial<IWalletTransactionApiResponse>): WalletsDetailsTransaction {
+    return new WalletsDetailsTransaction({
+      id: data.id ?? null,
+      amount: data.amount ?? 0,
+      creationDate: data.creationDate ?? '',
+      description: data.creationDate ?? '',
+      type: data.type ?? WalletTransactionType.Incomes,
+    });
+  }
+
+  public static createFromModalEditorData(id: number, data: ITransactionModalData): WalletsDetailsTransaction {
```

Comment:
Model should not be aware of some other elements inside app - like editors - if you need to create `WalletDetailsTransaction` you should do this part:
```typescript
new WalletsDetailsTransaction({
      id,
      amount: data.amount,
      creationDate: data.date.toString(),
      description: data.description ?? '',
      type: data.type,
    })
```
inside service or wherever you use `createFromModalEditorData`.